vars/checkReleaseIssues.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+import jenkins.ReleaseMetricsData
+ /**
+ * Library to check if release issue exists in the component repo.
+ * @param Map args = [:] args A map of the following parameters
+ * @param args.inputManifest <required> - Input manifest file(s) eg: [2.0.0/opensearch-2.0.0.yml, 2.0.0/opensearch-dashboards-2.0.0.yml] .
+ * @param args.action <optional> - Action to be performed. Default is 'check'. Acceptable values are 'check' and 'create'.
+ */
+void call(Map args = [:]) {
+    def inputManifest = args.inputManifest
+    String action = args.action ?: 'check'
+
+    // Parameter check
+    if (!inputManifest || inputManifest.isEmpty()) {
+        error "inputManifest parameter is required."
+    }
+    if (action != 'check' && action != 'create') {
+        error "Invalid action '${action}'. Valid values: check, create"
+    }
+
+    def inputManifestYaml = readYaml(file: args.inputManifest[0])
+    def version = inputManifestYaml.build.version
+    List<String> componentsMissingReleaseIssue = []
+
+    inputManifest.each { inputManifestFile ->
+        def inputManifestObj = readYaml(file: inputManifestFile)
```

Comment:
I believe we are reading the same file here for each iteration, can this be moved outside of the loop?